### PR TITLE
NEW new feature (select prospect status)

### PR DESCRIPTION
**Pull request by KiVoilaJe**
Change of the way to modify the status of the prospect by replacing it with a combobox.

Creation of a selectStatus function which allows to create a select with the different status, which then makes an ajax request to allow to modify the status without reload the page. In this function, we can see an if ($mode) which allows to call the js script only once by calling the SelectStatus function with the parameter ($mode = js).

### CODE REVIEW

File: htdocs/core/ajax/ajaxstatusprospect.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+$idstatus = GETPOST('id', 'int');
+$idprospect = GETPOST('prospectid', 'int');
+$action = GETPOST('action', 'aZ09');
+
+if ($action === "updatestatusprospect") {
+	$response = '';
+
+	$sql  = "UPDATE ".MAIN_DB_PREFIX."societe SET ";
+	$sql .= "fk_stcomm=".$db->escape($idstatus);
+	$sql .= " WHERE rowid = ".$db->escape($idprospect);
```
Comment (Hystepik) [inline htdocs/core/ajax/ajaxstatusprospect.php:57]: same as previous comment

File: htdocs/core/ajax/ajaxstatusprospect.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+if (!defined('NOREQUIRESOC'))   define('NOREQUIRESOC', '1');
+// If there is no need to load and show top and left menu
+if (!defined("NOLOGIN")) {
+	define("NOLOGIN", '1');
+}
+
+// Load Dolibarr environment
+require '../../main.inc.php';
+
+
```
Comment (eldy) [inline htdocs/core/ajax/ajaxstatusprospect.php:41]: Missing 
```
// Security check
if ($user->socid > 0) {
	unset($action);
	$socid = $user->socid;
}
$result = restrictedArea($user, 'societe', $object->id, '&societe');

```

File: htdocs/core/ajax/ajaxstatusprospect.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+if (!defined('NOREQUIREMENU'))  define('NOREQUIREMENU', '1');
+if (!defined('NOREQUIREHTML'))  define('NOREQUIREHTML', '1');
+if (!defined('NOREQUIREAJAX'))  define('NOREQUIREAJAX', '1');
+if (!defined('NOREQUIRESOC'))   define('NOREQUIRESOC', '1');
+// If there is no need to load and show top and left menu
+if (!defined("NOLOGIN")) {
+	define("NOLOGIN", '1');
+}
+
```
Comment (eldy) [inline htdocs/core/ajax/ajaxstatusprospect.php:37]: Add recuperation of paramaetr here

File: htdocs/core/ajax/ajaxstatusprospect.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * View
+ */
+
+top_httphead();
+
+$idstatus = GETPOST('id', 'int');
+$idprospect = GETPOST('prospectid', 'int');
+$action = GETPOST('action', 'aZ09');
+
+if ($action === "updatestatusprospect") {
```
Comment (eldy) [inline htdocs/core/ajax/ajaxstatusprospect.php:52]: if ($action === "updatestatusprospect" && $permisstiontoupdate) {

**Comment by eldy** [comment]
Add screenshot...

**Comment by FliyFly** [comment]
Here is an example of the select

**Comment by KiVoilaJe** [comment]
Here is an example of the selct: 
Capture d’écran du 2023-05-09 18-38-45

**Comment by Hystepik** [comment]
@eldy ok for me

**Pull request merged**